docs/inheritance_and_composition.md

```diff
@@ -0,0 +1,36 @@
+# Template Inheritance and Composition
+
+## What is Inheritance, what is Composition?
+
+Template Inheritance is a way to create a new template based on an existing one. It
+allows you to reuse code and structure from the parent template while adding or
+overriding specific parts in the child template. This is useful for creating variations
+of a template without duplicating code.
+
+<!-- TODO: add an example of inheritance -->
+
+Template Composition, on the other hand, is a way to combine multiple templates,
+independent of each other, into a single new template. In this way you can join the
+functionalities of multiple templates, by adding some "glue code", and avoid repetition.
```

Comment:
Do you have an approach like yours using tasks that run `copier copy` in mind? I've mostly thought of "composition" along the lines of https://copier.readthedocs.io/en/stable/configuring/#applying-multiple-templates-to-the-same-subproject. There, template composition means generating output from multiple templates in the same target directory, but there is no new template that is a composition of other templates.